packages/kmod-6.12-amdgpu/kmod-6.12-amdgpu.spec

```
@@ -0,0 +1,81 @@
+%global kernel_sources %{_cross_usrsrc}/kernels/6.12
+%define _kernel_version %(cat %{kernel_sources}/include/config/kernel.release)
+%global _cross_kmoddir %{_cross_libdir}/modules/%{_kernel_version}
+%global _ko ko
+
+# Cross-compilation make macro for kernel builds
+%global kmake %{shrink: \
+make -s \
+  ARCH="%{_cross_karch}" \
+  CROSS_COMPILE="%{_cross_target}-" \
+  INSTALL_HDR_PATH="%{buildroot}%{_cross_prefix}" \
+  INSTALL_MOD_PATH="%{buildroot}%{_cross_prefix}" \
+  INSTALL_MOD_STRIP=1 \
+  %{nil}}
+
+Name: %{_cross_os}kmod-6.12-amdgpu
+Version: 30.10
+Release: 1%{?dist}
+Summary: AMD GPU drivers for the 6.12 kernel
+License: GPL-2.0 WITH Linux-syscall-note
+URL: https://www.amd.com/
+
+Source0: https://repo.radeon.com/amdgpu/30.10.2/el/10/main/x86_64/amdgpu-dkms-6.14.14-2226257.el10.noarch.rpm
+
+BuildRequires: %{_cross_os}kernel-6.12-devel
+Requires: %{_cross_os}kernel-6.12
+
+%description
+%{summary}.
+
+%prep
+# Extract DKMS source from RPM
+rpm2cpio %{S:0} | cpio -idmu './usr/src/amdgpu-*'
+find usr/src/ -mindepth 1 -maxdepth 1 -type d -exec mv {} amdgpu \;
+rm -r usr
+
+%build
+# Build AMD GPU DKMS driver
+pushd %{_builddir}/amdgpu
+# Configure DKMS driver
+pushd amd/dkms
+sed -i -e 's,AC_TRY_COMMAND(make,AC_TRY_COMMAND(%kmake,g' m4/kernel.m4
+autoreconf -fi
+%cross_configure \
+  KERNELVER="%{_kernel_version}" \
+  ARCH="%{_cross_karch}" \
+  --with-linux=%{kernel_sources} \
+  %{nil}
+popd
+# Build using the AMD Makefile system
+%kmake \
+  KERNELVER="%{_kernel_version}" \
+  kernel_build_dir=%{kernel_sources} \
+  modules
+popd
+
+%install
+# Install AMD GPU kernel modules to correct location
+install -d %{buildroot}%{_cross_kmoddir}/kernel/drivers/gpu/drm/amd/amdgpu
+for module in \
+    amd/amdkcl/amdkcl \
+    ttm/amdttm \
+    amddrm_ttm_helper \
+    amddrm_buddy \
+    amddrm_exec \
+    scheduler/amd-sched \
+    amd/amdxcp/amdxcp \
+    amd/amdgpu/amdgpu; do
+  install -p -m 644 amdgpu/${module}.%{_ko} %{buildroot}%{_cross_kmoddir}/kernel/drivers/gpu/drm/amd/amdgpu/
```

Seems like this path will be more consistent with the NVIDIA drivers:
```suggestion
  install -p -m 644 amdgpu/${module}.%{_ko} %{buildroot}%{_cross_kmoddir}/kernel/drivers/extra/gpu/drm/amd/amdgpu
```